compaction/compaction_manager.cc

```diff
@@ -330,7 +330,7 @@ future<sstables::compaction_result> compaction_manager::task::compact_sstables(s
         if (release_exhausted) {
             release_exhausted(desc.old_sstables);
         }
-        t.on_compaction_completion(std::move(desc));
+        t.as_table_state().on_compaction_completion(std::move(desc), sstables::offstrategy::no).get();
```

Comment:
The task should work on the table_state / a compaction_group in the first place.
It shouldn't even know that a table has (currently) a `as_table_state` method.

Reply:
This is all removed later.You should have a look at the final state in the changed files tab.

Reply:
So maybe the patch series needs to go top down, using a back reference from table_state to table where needed, and removing those as the internal interfaces are converted to accepting table_state.

Reply:
Is a complete reorganization of 30 patches worth it to avoid a transient layering violation?

Reply:
IMO, yes.  We better add shims where need downstream, rather than temporarily change the api just for reverting it later.
It makes the series harder to reason about and to test.